src/algebra/free_algebra.lean

```diff
@@ -230,6 +227,34 @@ def lift (f : X → A) : free_algebra R X →ₐ[R] A :=
   map_add' := by { rintros ⟨⟩ ⟨⟩, refl },
   commutes' := by tauto }
 
+/--
+Given a function `f : X → A` where `A` is an `R`-algebra, `lift R f` is the unique lift
+of `f` to a morphism of `R`-algebras `free_algebra R X → A`.
+-/
+def lift : (X → A) ≃ (free_algebra R X →ₐ[R] A) :=
+{ to_fun := lift_aux R,
+  inv_fun := λ F, F ∘ (ι R),
+  left_inv := λ f, by {ext, refl},
+  right_inv := λ F, by {
+    ext x,
+    rcases x,
+    induction x,
+    case pre.of : {
+      change ((F : free_algebra R X → A) ∘ (ι R)) _ = _,
+      refl },
+    case pre.of_scalar : {
+      change algebra_map _ _ x = F (algebra_map _ _ x),
+      rw alg_hom.commutes F x, },
+    case pre.add : a b ha hb {
+      change lift_aux R (F ∘ ι R) (quot.mk _ _ + quot.mk _ _) = F (quot.mk _ _ + quot.mk _ _),
+      rw [alg_hom.map_add, alg_hom.map_add, ha, hb], },
+    case pre.mul : a b ha hb {
+      change lift_aux R (F ∘ ι R) (quot.mk _ _ * quot.mk _ _) = F (quot.mk _ _ * quot.mk _ _),
+      rw [alg_hom.map_mul, alg_hom.map_mul, ha, hb], }, }, }
+
```

Comment:
Let's make explicit that we don't want `lift_aux`:
```suggestion

@[simp] lemma lift_aux_eq (f : X → A) : lift_aux f = lift f := rfl
```

Reply:
> shouldn't some of these lifts also be a (linear?) (ring?) equiv?

Yes, almost certainly - I figured I'd leave that for a follow-up, if that's alright?

Reply:
If it requires a lot of work, don't feel like you have to do it. If the proof comes down to `by { unfold, simp }`, then I'd prefer to have it once and for all.

Reply:
Actually, I think the answer is no - `alg_hom` has no `add` defined, so the equivalence can't be additive, and I can't think of anything else it could be either.

Reply:
Ah of course, we wouldn't preserve the image of `R` if we started adding/multiplying `alg_hom`s.